src/jdk.compiler/share/classes/com/sun/tools/javac/tree/DocPretty.java

```diff
@@ -76,6 +76,12 @@ protected void print(Object s) throws IOException {
         out.write(Convert.escapeUnicode(s.toString()));
     }
 
+    /** Print character. Should be only used internally for known ASCII characters.
+     */
+    private void print(char c) throws IOException {
+        out.write(c);
```

Comment:
The restriction to skip calling `Convert.escapeUnicode(s.toString())` leaks into `print(List, char sep)`.

If we wanted to be paranoid we could do one or more of the following;

* Change the body of this method to something like<br> `if (Character.isASCII(c)) out.write(c); else print((Character) c);`
* include an `assert` or call to a method in `Assert.*` to verify the argument is ASCII.
* change that `protected void print(List, char)` method to be `private` instead of `protected`

Reply:
Already asked above: should we really be paranoid here? There is a big class comment above that anyone depending on the internal api uses this at his own risk...
But if I should change anything I'd rather make the method `private`

Reply:
_(Sorry for the duplicate entry)_

I think making the method `private` is good enough, and doing it in the same changeset that changes the signature is even better for future code historians.